models/app.models.js

```diff
@@ -21,4 +21,26 @@ exports.fetchArticleById = async id => {
 	} else {
 		return results.rows[0];
 	}
-};
\ No newline at end of file
+};
+
+exports.fetchArticles = async () => {
+	const articleQuery = await db.query(
+		'SELECT * FROM articles ORDER BY created_at DESC'
+	);
+	const commentQuery = await db.query('SELECT * FROM comments');
+
+	if (articleQuery.rowCount === 0) {
+		return Promise.reject({ code: 404 });
+	} else {
+		const { rows: articles } = articleQuery;
+		const { rows: comments } = commentQuery;
+
+		articles.map(article => {
+			article['comment_count'] = comments.filter(
+				comment => comment.article_id === article.article_id
+			).length;
```

Comment:
This is something that is possible to do with one SQL query string, by making use of `COUNT`, `JOIN`, and `ORDER BY` within the query. I feel like doing this with nested loops as you have, may be missing out on an opportunity to show off your SQL skills to potential employers

Reply:
my first thoughts were to go straight to js, i am not very confident with SQL but i think i could give it a try